### PR TITLE
Fix LK catalyst shared object copy linux/mac only

### PR DESCRIPTION
**Pull request by josephleekl**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
This is a fix for [this PR](https://github.com/PennyLaneAI/pennylane-lightning/pull/945) failing lightning kokkos build for non-linux systems.

[sc-76296]
**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by multiphaseCFD** [comment]
Thanks @josephleekl ! LGTM. Don't forget to add an entry to the changelog and create a SC story for this PR and link the SC story to it.

**Review by maliasadi — APPROVED** [review]
Thanks @josephleekl :raised_hands:

**Review by multiphaseCFD — APPROVED** [review]
I'm second to @maliasadi on limit the copy to linux system

File: setup.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def build_extension(self, ext: CMakeExtension):
         )
 
         # Ensure that catalyst shared object is copied to the build directory for pip editable install
-        if backend in ("lightning_kokkos"):
+        if platform.system() != "Darwin" and backend in ("lightning_kokkos"):
```
[the remark this answers is not in the document]
Comment (josephleekl) [reply setup.py:160]: That's a good idea

**Comment by multiphaseCFD** [comment]
BTW, would you like to add the `ci_build_wheel` label to ensure all CIs could pass? You can re-trigger the CI after choosing the label.

File: setup.py [hunk 2]
```diff
@@ -157,7 +157,7 @@ def build_extension(self, ext: CMakeExtension):
         )
 
         # Ensure that catalyst shared object is copied to the build directory for pip editable install
-        if backend in ("lightning_kokkos"):
+        if platform.system() == "Linux" and backend in ("lightning_kokkos"):
             source = os.path.join(f"{extdir}", f"lib{backend}_catalyst.so")
             destination = os.path.join(os.getcwd(), "build")
             shutil.copy(source, destination)
```
Comment (AmintorDusko) [inline setup.py:163]: Do we have any problem with building shared libraries for Mac? If not, we could check for Mac and move the `.dylib` file. Maybe some similar for Windows? (I forgot if this was supposed to build for Windows.)

Comment (AmintorDusko) [reply setup.py:163]: I mean, Catalyst will not work with Windows, but it is prepared to look for shared libraries in Mac systems.

Comment (josephleekl) [reply setup.py:163]: It should build shared libraries for Mac too, I can check if moving `.dylib` instead of `.so` works

Comment (josephleekl) [reply setup.py]: On Mac it works with copying dylib, I've made a new commit now

File: setup.py [hunk 3]
```diff
@@ -158,9 +158,14 @@ def build_extension(self, ext: CMakeExtension):
...
-            destination = os.path.join(os.getcwd(), "build")
-            shutil.copy(source, destination)
+            if platform.system() == "Linux":
+                source = os.path.join(f"{extdir}", f"lib{backend}_catalyst.so")
+                destination = os.path.join(os.getcwd(), "build")
+                shutil.copy(source, destination)
+            if platform.system() == "Darwin":
+                source = os.path.join(f"{extdir}", f"lib{backend}_catalyst.dylib")
+                destination = os.path.join(os.getcwd(), "build")
+                shutil.copy(source, destination)
```
Comment (AmintorDusko) [inline setup.py:168]:
```suggestion
            if platform.system() in ["Linux", "Darwin"]:
                shared_lib_ext = {"Linux": ".so", "Darwin": ".dylib"}
                source = os.path.join(f"{extdir}", f"lib{backend}_catalyst{shared_lib_ext}")
                destination = os.path.join(os.getcwd(), "build")
                shutil.copy(source, destination)
```
I'm assuming here that you tested it both for Linux and Mac systems.

Comment (AmintorDusko) [reply setup.py:168]: wait!

Comment (AmintorDusko) [reply setup.py:168]: OK, you can proceed now.

File: setup.py (hunk 2 above)
Comment (AmintorDusko) [reply setup.py]: Great! Thank you so much for checking that!

File: setup.py (hunk 3 above)
Comment (josephleekl) [reply setup.py:168]: Yes it's tested for Linux and Mac, thanks @AmintorDusko!

**Review by AmintorDusko — APPROVED** [review]
Great job!

**Pull request merged**